uniffi_bindgen/src/templates/ObjectTemplate.rs

```diff
@@ -48,6 +48,42 @@ uniffi::deps::lazy_static::lazy_static! {
         uniffi::deps::log::debug!("{{ meth.ffi_func().name() }}");
         // If the method does not have the same signature as declared in the UDL, then
         // this attempt to call it will fail with a (somewhat) helpful compiler error.
+        {% if meth.is_static() %}
+        {% call rs::to_rs_static_method_call(obj, meth) %}
+        {% else %}
         {% call rs::to_rs_method_call(obj, meth) %}
+        {% endif %}
     }
 {% endfor %}
+
+// We proide a `ViaFfi` impl for each object, but the only thing it can currently
+// do is lower an owned instance into a handle. This is useful for returning new
+// instances of the object from static methods, and we know it is safe and sound
+// because Rust's ownership system ensures that:
+//  * the thing we're operating on is an owned instance (not a reference) and
+//    so we know that nothing else in the program is looking at it.
+//  * the ownership is irrevokably transferred to the handlemap, and hence to
+//    the foreign language code, which takes responsibility for it from there.
+//
+// Other operations are not yet supported, since they would involve much more
+// complicated semantics around references.
+unsafe impl uniffi::ViaFfi for {{ obj.name() }} {
+    type FfiType = u64;
+
+    fn lower(self) -> Self::FfiType {
+        // Note that this consumes `self`, transferring ownership to the handlemap.
+        {{ handle_map }}.insert(self).into_u64()
+    }
```

Comment:
I haven't developed the ownership mental model enough to know if this is true: my suspicion is that we need to do a bit more error reporting, or book keeping to make sure this doesn't get misused: either deliberately or not.

```rust
fn get_self(self) -> Self { self }
```

```kotlin
val thing1 = Thing()
val thing2 = thing1.getSelf()

assert thing1 == thing2 // fail!
```